needle/core/utils/utils.py

```diff
@@ -40,6 +40,11 @@ def extract_paths_from_string(str):
         # Error
         return None, None
 
+    @staticmethod
+    def get_file_path(file_str):
+        # Returns the irectory of a given full file path.
+        return '/'.join(file_str.split('/')[:-1])

```

Comment:
you should use a more "pythonic" way.
Have you tried this: `os.path.dirname(fullpath)` ?